lib/common_test/src/ct_logs.erl

```diff
@@ -1385,31 +1389,45 @@ make_last_run_index1(SuiteName, [LogDir | LogDirs], Result, TotSucc, TotFail,
 	    make_last_run_index1(SuiteName, LogDirs, [Result|Result1],
 				 TotSucc+Succ, 
 				 TotFail+Fail, UserSkip+USkip, AutoSkip1,
-				 TotNotBuilt+NotBuilt, Missing);
+				 TotNotBuilt+NotBuilt, TotElapsedTime+ElapsedTime,
+				 Missing);
 	error ->
 	    make_last_run_index1(SuiteName, LogDirs, Result, TotSucc, TotFail,
-				 UserSkip, AutoSkip, TotNotBuilt, Missing)
+				 UserSkip, AutoSkip, TotNotBuilt, TotElapsedTime, Missing)
     end;
 make_last_run_index1(_, [], Result, TotSucc, TotFail,
-		     UserSkip, AutoSkip, TotNotBuilt, _) ->
-    {Result,TotSucc,TotFail,UserSkip,AutoSkip,TotNotBuilt}.
+		     UserSkip, AutoSkip, TotNotBuilt, TotElapsedTime, _) ->
+    {Result,TotSucc,TotFail,UserSkip,AutoSkip,TotNotBuilt,TotElapsedTime}.
 
 make_one_index_entry(SuiteName, LogDir, Label, All, Missing, URIs) ->
-    case count_cases(LogDir) of
-	{Succ,Fail,UserSkip,AutoSkip} ->
+    case {All, count_cases(LogDir)} of
+	{false, {Succ,Fail,UserSkip,AutoSkip,ElapsedTime}} ->
 	    NotBuilt = not_built(SuiteName, LogDir, All, Missing),
+		%% All = false, meaning we're generating ct_logs/ct_run.*/index.html
+		%% meaning index.html of one specific execution
+		%% We pass ElapsedTime there
 	    {NewResult,URIs1} = make_one_index_entry1(SuiteName, LogDir, Label,
 						      Succ, Fail,
 						      UserSkip, AutoSkip,
 						      NotBuilt, All,
-						      normal, URIs),
-	    {NewResult,Succ,Fail,UserSkip,AutoSkip,NotBuilt,URIs1};
-	error ->
+						      normal, URIs, ElapsedTime),
+	    {NewResult,Succ,Fail,UserSkip,AutoSkip,NotBuilt,URIs1,ElapsedTime};
+	{_, {Succ,Fail,UserSkip,AutoSkip,ElapsedTime}} ->
+		NotBuilt = not_built(SuiteName, LogDir, All, Missing),
+		%% All = {true, _} meaning we're generating ct_logs/index.html
+		%% No ElapsedTime is required there, we pass undefined as ElapsedTime
+	    {NewResult,URIs1} = make_one_index_entry1(SuiteName, LogDir, Label,
+						      Succ, Fail,
+						      UserSkip, AutoSkip,
+						      NotBuilt, All,
+						      normal, URIs, undefined),
+	    {NewResult,Succ,Fail,UserSkip,AutoSkip,NotBuilt,URIs1,ElapsedTime};
+	{_, error} ->
```

Comment:
can you rewrite that part in a way that we minimize amount of redundant code and optimize ease of reading?
maybe by having new small anonymous helper function defined before existing case?

```
MaybeAddElapsedTime =
fun(_All = false, ElapsedTime) -> ElapsedTime;
   (_, _) -> undefined
end
```

code comments you added are nice, but maybe with anonymous function above it would be enough to explain what All = false mean. IMHO other stuff becomes more obvious ... what do you think?